File: Two Pointer/1968. Array With Elements Not Equal to Average of Neighbors/1968. Array With Elements Not Equal to Average of Neighbors.py
# Problem          : 1968. Array With Elements Not Equal to Average of Neighbors
# Time Complexity  : O(n)
# Python Solution

class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        n = len(nums)

        for i in range(1, n - 1):
            if 2 * nums[i] == (nums[i - 1] + nums[i + 1]):
                nums[i], nums[i + 1] = nums[i + 1], nums[i]

        for i in range(n - 2, 0, -1):
            if 2 * nums[i] == (nums[i - 1] + nums[i + 1]):
                nums[i], nums[i - 1] = nums[i - 1], nums[i]

        return nums
    
    

File: Two Pointer/1968. Array With Elements Not Equal to Average of Neighbors/1968. Array With Elements Not Equal to Average of Neighbors.cpp
/*
# Problem          : 1968. Array With Elements Not Equal to Average of Neighbors
# Time Complexity  : O(n)
# C++ Solution
*/

class Solution {
public:
    vector<int> rearrangeArray(std::vector<int>& nums) {
        int n = nums.size();

        for (int i = 1; i < n - 1; i++) {
            if (2 * nums[i] == (nums[i - 1] + nums[i + 1])) {
                std::swap(nums[i], nums[i + 1]);
            }
        }
        for (int i = n - 2; i > 0; i--) {
            if (2 * nums[i] == (nums[i - 1] + nums[i + 1])) {
                swap(nums[i], nums[i - 1]);
            }
        }
        return nums;
    }
};